python-threatexchange/threatexchange/cli/match_cmd.py

```diff
@@ -215,6 +221,90 @@ def execute(self, settings: CLISettings) -> None:
                             fetched_data,
                         )
 
+    def validate(
+        self,
+        settings: CLISettings,
+        hashes_grouped_by_prefix: t.Dict[t.Optional[str], t.Set[str]],
+        signal_types: t.List[t.Type[SignalType]],
+    ) -> t.Dict[t.Type[SignalType], t.Set[str]]:
+        """
+        Takes the hashes grouped by optional string prefix, performs all required validations.
+        Command line arguments, SignalType, hashes, and ambiguous SignalTypes are validated.
+        Returns a dict of validated SignalType, and validated hashes of each SignalType.
+        """
+
+        # There is a signal without a prefix and it's ambiguous (fix: -S)
+        if None in hashes_grouped_by_prefix:
+            if len(hashes_grouped_by_prefix) > 2 or len(signal_types) > 1:
```

Comment:
ignorable: I think this is being overly specific about the input. This will throw if you do

```
cat 'pdq facefaceface...
facefaceface...' > file.txt
$ tx match -S pdq photo file.txt
```
but I think it's unlikely that you'll get a non-prefixed value here that will pass the signal type validation on L247

Reply:
ohhh, this is a good catch. I see what you mean. Thanks!